src/conversions/chrono.rs

```diff
@@ -200,34 +200,24 @@ impl<Tz: TimeZone> IntoPy<PyObject> for DateTime<Tz> {
     }
 }
 
-impl FromPyObject<'_> for DateTime<FixedOffset> {
-    fn extract(ob: &PyAny) -> PyResult<DateTime<FixedOffset>> {
+impl<Tz: TimeZone + for<'a> FromPyObject<'a>> FromPyObject<'_> for DateTime<Tz> {
+    fn extract(ob: &PyAny) -> PyResult<DateTime<Tz>> {
         let dt: &PyDateTime = ob.downcast()?;
-        let tz: FixedOffset = if let Some(tzinfo) = dt.get_tzinfo() {
+        let tz = if let Some(tzinfo) = dt.get_tzinfo() {
             tzinfo.extract()?
         } else {
             return Err(PyTypeError::new_err(
                 "expected a datetime with non-None tzinfo",
             ));
         };
         let dt = NaiveDateTime::new(py_date_to_naive_date(dt)?, py_time_to_naive_time(dt)?);
-        // `FixedOffset` cannot have ambiguities so we don't have to worry about DST folds and such
-        Ok(dt.and_local_timezone(tz).unwrap())
-    }
-}
-
-impl FromPyObject<'_> for DateTime<Utc> {
-    fn extract(ob: &PyAny) -> PyResult<DateTime<Utc>> {
-        let dt: &PyDateTime = ob.downcast()?;
-        let _: Utc = if let Some(tzinfo) = dt.get_tzinfo() {
-            tzinfo.extract()?
-        } else {
-            return Err(PyTypeError::new_err(
-                "expected a datetime with non-None tzinfo",
-            ));
-        };
-        let dt = NaiveDateTime::new(py_date_to_naive_date(dt)?, py_time_to_naive_time(dt)?);
-        Ok(dt.and_utc())
+        dt.and_local_timezone(tz).single().ok_or_else(|| {
+            PyValueError::new_err(format!(
+                "The datetime {} contains an incompatible or ambiguous timezone",
+                ob.repr()
```

Comment:
You could use the built-in `Debug/repr` or `Display/str` implementations for `PyAny` which already handle fallbacks.

Reply:
They are defined in

https://github.com/PyO3/pyo3/blob/ff50285d1f47327d8cd549730e24dd62944caf6c/src/types/mod.rs#L94-117

if you want to check whether they work for you.

Reply:
Thank you! Done.

Reply:
I allowed myself to update an other `.repr()` call in the file